liblwgeom/topo/lwgeom_topo.c

```diff
@@ -2141,9 +2141,8 @@ _lwt_AddFaceSplit( LWT_TOPOLOGY* topo,
         return -2;
       }
 
-      /* IDEA: check that bounding box shortcut is taken, or use
-       *       shellbox to do it here */
-      contains = ptarray_contains_point(pa, &ep);
+      contains = gbox_contains_point2d(newface.mbr, &ep) == LW_TRUE ? LW_INSIDE : LW_OUTSIDE;
```

Comment:
should you use "shellbox" here instead of "newface.mbr" ? Just to follow the IDEA comment, didn't really check the code (yes I know it's hard to follow)

Reply:
Ah yes :) seems to work better now. In a case of loading a pretty large topology via first adding nodes then all edges with ST_AddEdgeNewFaces in ported code it speeds up the process by more than 10x (!).